test/link/names.txt

```diff
@@ -39,45 +41,48 @@ Type:
  - [2] (i64) -> nil
  - [3] (i32) -> nil
 Import:
- - func[0] sig=0 <- __extern.missing
+ - func[0] sig=0 <- __extern.missing0
+ - func[1] sig=0 <- __extern.missing1
+ - func[2] sig=0 <- __extern.missing2
 Function:
- - func[1] sig=1
- - func[2] sig=2
- - func[3] sig=2
- - func[4] sig=3
+ - func[3] sig=1
+ - func[4] sig=2
+ - func[5] sig=2
+ - func[6] sig=3
 Export:
- - func[1] foo
- - func[4] baz
+ - func[3] foo
+ - func[6] baz
 Custom:
  - name: "name"
- - func[0] $import_func2
- - func[1] $name1
- - func[2] $name2
- - func[4] $name3
+ - func[0] $import_func0
+ - func[2] $import_func2
+ - func[3] $name1
+ - func[4] $name2
+ - func[6] $name3
 Custom:
  - name: "reloc.Code"
   - section: Code
-   - R_FUNC_INDEX_LEB   idx=0    addend=0    offset=0x6(file=0x5a)
-   - R_FUNC_INDEX_LEB   idx=0x1  addend=0    offset=0x11(file=0x65)
-   - R_FUNC_INDEX_LEB   idx=0    addend=0    offset=0x1f(file=0x73)
+   - R_FUNC_INDEX_LEB   idx=0x2  addend=0    offset=0x6(file=0x83)
+   - R_FUNC_INDEX_LEB   idx=0x3  addend=0    offset=0x11(file=0x8e)
+   - R_FUNC_INDEX_LEB   idx=0    addend=0    offset=0x1f(file=0x9c)
 
 Code Disassembly:
 
-000055 <$name1>:
- 000057: 41 01                      | i32.const 0x1
- 000059: 10 84 80 80 80 00          | call 0x4
-           00005a: R_FUNC_INDEX_LEB   0
- 00005f: 0b                         | end
-000060 <$name2>:
- 000062: 42 01                      | i64.const 1
- 000064: 10 81 80 80 80 00          | call 0x1
-           000065: R_FUNC_INDEX_LEB   1
- 00006a: 0b                         | end
-00006b func[3]:
- 00006d: 0b                         | end
-00006e <$name3>:
- 000070: 41 02                      | i32.const 0x2
- 000072: 10 84 80 80 80 00          | call 0x4
-           000073: R_FUNC_INDEX_LEB   0
- 000078: 0b                         | end
+00007e <$name1>:
+ 000080: 41 01                      | i32.const 0x1
+ 000082: 10 86 80 80 80 00          | call 0x6
+           000083: R_FUNC_INDEX_LEB   2
+ 000088: 0b                         | end
+000089 <$name2>:
+ 00008b: 42 01                      | i64.const 1
+ 00008d: 10 83 80 80 80 00          | call 0x3
+           00008e: R_FUNC_INDEX_LEB   3
```

Comment:
I think this line is wrong, and should be 0x2 (which it is with the possibly fixed `relocate_func_index` I'm trying), but I'd appreciate a second opinion. $name2 calls index 3 in the original binary, which is $import_func_2, which is func[2] in the linked binary. Right?